rust/arrow/src/json/reader.rs

```diff
@@ -1559,6 +1608,54 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_nested_struct_json_arrays() {
```

Comment:
This is a pretty cool test case demonstrating what is going on -- TIL I learn that we have a JSON schema extractor in arrow!